.github/workflows/ci.yml

```
@@ -157,6 +157,25 @@ jobs:
       - name: Perform CodeQL Analysis
         uses: github/codeql-action/analyze@a34ca99b4610d924e04c68db79e503e1f79f9f02 # v2
 
+  trivy-scan-fs:
+    runs-on: ubuntu-22.04
+    needs: detect-noop
+    if: needs.detect-noop.outputs.noop != 'true'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@24cb9080177205b6e8c946b17badbe402adc938f # v3
+        with:
+          submodules: true
+
+      - name: Run Trivy vulnerability scanner in fs mode
+        uses: aquasecurity/trivy-action@1f0aa582c8c8f5f7639610d6d38baddfea4fdcee # 0.9.2
+        with:
+          scan-type: 'fs'
+          ignore-unfixed: true
+          skip-dirs: design
+          scan-ref: '.'
+          severity: 'CRITICAL,HIGH'
```

Do you think if it makes sense to exit 1 in case of critical? I am a bit worried that information is lost when we have a green check at the end?

---

Not sure I got your suggestion. This setting means that it is going to exit with exit code 1 in case of critical or high vulnerabilities detected.

---

Not high but critical. 

AFAIU, it currently outputs the scan result and always return success no matter what it found, right?
I am wondering if it makes sense to exit 1 in case of `critical` one to make it more visible.

---

right, yes, I'd leave it for both HIGH and CRITICAL:
```suggestion
          exit-code: '1'
          severity: 'CRITICAL,HIGH'
```

---

I am not sure about `high`. 
For example, right now there is 1 high, and CI will be failing until we resolve it. It could be reasonable when there is a `critical` one may be.

```
Total: 1 (HIGH: 1, CRITICAL: 0)

┌──────────────────┬────────────────┬──────────┬───────────────────┬───────────────┬─────────────────────────────────────────────────────────────┐
│     Library      │ Vulnerability  │ Severity │ Installed Version │ Fixed Version │                            Title                            │
├──────────────────┼────────────────┼──────────┼───────────────────┼───────────────┼─────────────────────────────────────────────────────────────┤
│ golang.org/x/net │ CVE-2022-41723 │ HIGH     │ 0.4.0             │ 0.7.0         │ golang.org/x/net/http2: avoid quadratic complexity in HPACK │
│                  │                │          │                   │               │ decoding                                                    │
│                  │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2022-41723                  │

```

---

we should fix it then :wink: https://github.com/crossplane/crossplane-runtime/pull/388